drivers/ukrandom/lcpu/arch/x86_64/random.c

```diff
@@ -0,0 +1,149 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <errno.h>
+#include <uk/arch/lcpu.h>
+#include <uk/arch/types.h>
+#include <uk/errptr.h>
+#include <uk/print.h>
+#include <uk/random/driver.h>
+
+/* Intel Digital Random Number Generator (DRNG)
+ * Sect. 5.2.1 Retry Recommendations
+ */
+#define RDRAND_RETRY_LIMIT	10
+
+static __bool have_rdseed;
+
+static inline int rdrand(__u64 *val)
+{
+	__u8 res;
+
+	for (int i = 0; i < RDRAND_RETRY_LIMIT; i++) {
+		__asm__ __volatile__(
+			"	rdrand	%0\n" /* get rand */
+			"	setc	%1\n" /* get result */
+			: "=r" (*val), "=qm" (res)
+		);
+		if (unlikely(!res))
+			return -EIO;
+	}
+
+	return 0;
+}
+
+static inline int rdseed(__u64 *val)
+{
+	__u8 res;
+
+	if (!have_rdseed)
+		return -ENOTSUP;
```

Comment:
The commit message states that **-ENODEV** is returned if RDSEED is not supported. I think the `-ENOTSUP` is more appropriate and the commit message should be updated.